src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,24 +1,39 @@
-# TO-DO: Complete the selection_sort() function below
+import time
+
+# * Complete the selection_sort() function below
+# ? - O(n²)
+
 def selection_sort(arr):
-    # loop through n-1 elements
+
+    # * loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc)
+        # * find next smallest element
+        # * (hint, can do in 3 loc)
         # Your code here
+        for j in range(cur_index + 1, len(arr)):
+            if arr[smallest_index] > arr[j]:
+                smallest_index = j
+
 
 
-        # TO-DO: swap
+        # * swap
         # Your code here
+        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]
 
     return arr
```

Comment:
This is a great implementation! Having said that, there's some duplication that can be DRY'd up to provide you with a slightly more Pythonic and readable solution.

For starters, alter the `for` loop provided to you to use `enumerate(arr)` rather than `range(len(arr)-1)`.

Enumerate returns a tuple containing the current index and the item at that index, which you can unpack like so:

```python
for index, item in enumerate(arr)
```

From there, you can store a _single_ additional mutable memory reference for the index, called something like `target`, and representing the current target index for the inner for loop.

The rest of the code looks roughly identical from there, which brings the total function to the following:

```python
def selection_sort(arr):
    for index, item in enumerate(arr):
        target = index

        for x in range(index+1, len(arr)):
            if arr[x] < arr[target]:
                target = x

        arr[index], arr[target] = arr[target], item

    return arr
```

If you _really_ wanted to to be extra, you could add typing annotations to make your editor and your brain happy:

```python
from typing import List

def selection_sort(arr: List[int]) -> List[int]:
    for index, item in enumerate(arr):
        target: int = index

        for x in range(index+1, len(arr)):
            if arr[x] < arr[target]:
                target = x

        arr[index], arr[target] = arr[target], item

    return arr